core/src/main/java/io/grpc/transport/AbstractStream.java

```diff
@@ -289,6 +296,26 @@ protected final void requestMessagesFromDeframer(int numMessages) {
   }
 
   /**
+   * Event handler to be called by the subclass when the stream's headers have passed any connection
+   * flow control (i.e., MAX_CONCURRENT_STREAMS). It may call the listener's {@link
+   * StreamListener#onReady()} handler if appropriate. This must be called from the transport
+   * thread, since the listener may be called back directly.
+   */
+  protected void onStreamAllocated() {
```

Comment:
Do we still need `notifyIfReady`? IIRC this was the only time it was used by the subclasses.